cinema/views.py

```diff
@@ -118,21 +149,27 @@ class OrderPagination(PageNumberPagination):
     max_page_size = 100
 
 
-class OrderViewSet(viewsets.ModelViewSet):
+class OrderViewSet(
+    mixins.ListModelMixin, mixins.CreateModelMixin, viewsets.GenericViewSet
+):
     queryset = Order.objects.prefetch_related(
         "tickets__movie_session__movie", "tickets__movie_session__cinema_hall"
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    authentication_classes = [TokenAuthentication]
+    permission_classes = [IsAuthenticated]
 
     def get_queryset(self):
-        return Order.objects.filter(user=self.request.user)
+        queryset = Order.objects.prefetch_related(
+            "tickets__movie_session__movie",
+            "tickets__movie_session__cinema_hall",
+        )
+        if self.request.user.is_staff:
+            queryset = queryset.filter(user=self.request.user)
+        return queryset
```

Comment:
There is a logical error in the `get_queryset` method of the `OrderViewSet`. The current logic filters orders for staff users only, which seems incorrect. Typically, staff users should have access to all orders, while non-staff users should only see their own orders. Consider revising the logic to reflect this.